pkg/common/consumer/consumer_factory.go

```diff
@@ -64,13 +66,17 @@ func (c *customConsumerGroup) Close() error {
 var _ sarama.ConsumerGroup = (*customConsumerGroup)(nil)
 
 // StartConsumerGroup creates a new customConsumerGroup and starts a Consume goroutine on it
-func (c kafkaConsumerGroupFactoryImpl) StartConsumerGroup(groupID string, topics []string, logger *zap.SugaredLogger, handler KafkaConsumerHandler, options ...SaramaConsumerHandlerOption) (sarama.ConsumerGroup, error) {
+func (c kafkaConsumerGroupFactoryImpl) StartConsumerGroup(ctx context.Context, groupID string, topics []string, handler KafkaConsumerHandler, options ...SaramaConsumerHandlerOption) (sarama.ConsumerGroup, error) {
+	logger := logging.FromContext(ctx)
+
 	consumerGroup, err := c.createConsumerGroup(groupID)
 	if err != nil {
+		logger.Errorw("Unable to create consumer group", zap.Error(err))
```

Comment:
NIT: error message starting with lower case

Reply:
hrm - not sure what you mean here

Reply:
I mean something like
```go
logger.Errorw("Unable to create consumer group", zap.Error(err), zap.Sting("Consumer Group", groupID))
```